jekyll/_cci2/env-vars.md

```diff
@@ -47,14 +47,63 @@ Running scripts within configuration
 may expose secret environment variables.
 See the [Using Shell Scripts]({{ site.baseurl }}/2.0/using-shell-scripts/#shell-script-best-practices) document for best practices for secure scripts.
 
-## Setting an Environment Variable in a Shell Command
+### Using `BASH_ENV` to Set Environment Variables
 
 CircleCI does not support interpolation
-when defining configuration variables like `working_directory` or `images`.
+when setting environment variables.
 All defined values are treated literally.
+This can cause issues
+when defining `working_directory`,
+modifying `PATH`,
+and sharing variables across multiple `run` steps.
+
+In the example below,
+`$ORGNAME` and `$REPONAME` will not be interpolated.
+
+```yaml
+working_directory: /go/src/github.com/$ORGNAME/$REPONAME
+```
+
+As a workaround,
+use a `run` step
+to export environment variables to `BASH_ENV`,
+as shown below.
+
+```yaml
+steps:
+  - run:
+      name: Setup Environment Variables
+      command: |
+        echo 'export PATH="$GOPATH/bin:$PATH"' >> $BASH_ENV
+        echo 'export GIT_SHA1="$CIRCLE_SHA1"' >> $BASH_ENV
+```
 
-However, it is possible to interpolate a variable within a command
-by setting it for the current shell.
+In every step,
+CircleCI automatically loads and runs `BASH_ENV`,
```

Comment:
In shells, we would say that, "Bash sources `BASH_ENV`". I wouldn't leave out your explanation though because not everyone would know what that means. Perhaps combine the two?